packages/@react-spectrum/color/stories/HexColorField.stories.tsx

```diff
@@ -0,0 +1,136 @@
+ */
+
+import {action} from '@storybook/addon-actions';
+import {ActionButton} from '@react-spectrum/button';
+import {Color} from '@react-stately/color';
+import {Dialog, DialogTrigger} from '@react-spectrum/dialog';
+import {Flex} from '@react-spectrum/layout';
+import {HexColorField} from '../';
+import React, {useState} from 'react';
+import {storiesOf} from '@storybook/react';
+import {View} from '@react-spectrum/view';
+
+storiesOf('HexColorField', module)
+  .add(
+    'Default',
+    () => render()
+  )
+  .add(
+    'has default value',
+    () => render({defaultValue: '#abcdef'})
+  )
+  .add(
+    'isQuiet',
+    () => render({isQuiet: true})
+  )
+  .add(
+    'isReadOnly',
+    () => render({isReadOnly: true, defaultValue: '#abcdef'})
+  )
+  .add(
+    'isDisabled',
+    () => render({isDisabled: true, defaultValue: '#abcdef'})
+  )
+  .add(
+    'validationState valid',
+    () => render({validationState: 'valid'})
+  )
+  .add(
+    'validationState invalid',
+    () => render({validationState: 'invalid'})
+  )
+  .add(
+    'with placeholder',
+    () => render({placeholder: 'Enter a hex color'})
+  )
+  .add(
+    'step = 16',
+    () => render({step: 16})
+  )
+  .add(
+    'controlled value',
+    () => (
+      <ControlledHexColorField
+        value={new Color('#FF00AA')}
+        onChange={action('change')} />
+    )
+  )
+  .add(
+    'as a popover',
+    () => (
+      <HexColorFieldPopover
+        label="Choose a color"
+        value="#ff0000"
+        step={255}
+        onChange={action('change')} />
+    )
+  )
+  .add(
+    'as a popover, defaults only',
+    () => <HexColorFieldPopover onChange={action('change')} />
+  );
+
+function HexColorFieldPopover(props: any = {}) {
+  let [color, setColor] = useState(props.value);
+  let colorString = color ? color.toString('hex') : '';
+  return (
+    <DialogTrigger type="popover">
+      <ActionButton
+        width="size-1600"
+        height="size-1600"
+        UNSAFE_style={{
+          background: colorString
+        }} >{colorString}</ActionButton>
+      <Dialog 
+        width="size-3600"
+        height="size-1600" >
+        <View padding="size-300">
+          {render({
+            ...props,
+            value: color,
+            onChange: (newColor: Color) => {
+              setColor(newColor);
+              if (props.onChange) { props.onChange(newColor); }
+            }
+          })}
+        </View>
+      </Dialog>
+    </DialogTrigger>
+  );
+}
+
+function ControlledHexColorField(props: any = {}) {
+  let [color, setColor] = useState(props.value);
+  let onChange = (color: Color) => {
+    setColor(color);
+    if (props.onChange) { props.onChange(color); }
+  };
+  return (
+    <Flex direction="row" gap="size-100" alignItems="end">
+      <HexColorField
+        label="Primary Color"
+        onChange={onChange}
+        value={color} />
+      <View width="size-400" height="size-400" UNSAFE_style={{backgroundColor: color.toString('rgb')}} />
```

Comment:
Hi @snowystinger, regarding

> Found an issue with the controlled state, if I delete the text, the story throws an error and doesn't recover.
One thing to keep in mind while fixing that issue is that whatever comes out of onChange must be able to be interpreted by 'value' so that controlled can work. My guess is that value can't accept null.
This is a tricky edge case, lets make sure to add a test for it, you've been amazing with all the tests you've added so far, it makes my job much easier.

I feel like the issue is with the story and not with the code. It doesn't check for an empty/falsey value for `color` and assumes all values of `color` will have methods like `toString('rgb')` that is throwing the error. The story would fail as well if it didn't have a `value` prop provided to it.

```
  .add(
    'controlled value',
    () => (
      <ControlledHexColorField
        value={new Color('#FF00AA')}     // when taken out, you get "Cannot read property 'toString' of undefined"
        onChange={action('change')} />
    )
  )
```

My proposed fix is to have a check first in the story before calling `Color`-specific methods.

```suggestion
function ControlledHexColorField(props: any = {}) {
  let [color, setColor] = useState(props.value);
  let onChange = (color: Color) => {
    setColor(color);
    if (props.onChange) { props.onChange(color); }
  };
  let style = color ? {backgroundColor: color.toString('rgb')} : {};
  return (
    <Flex direction="row" gap="size-100" alignItems="end">
      <HexColorField
        label="Primary Color"
        onChange={onChange}
        value={color} />
      <View width="size-400" height="size-400" UNSAFE_style={style} />
    </Flex>
  );
}
```

Reply:
o, i see, that's great news :)
then yes, I think what you're proposing is a fine fix for the story